app/components/channel_drawer/channels_list/list/list.js

```diff
@@ -244,24 +272,33 @@ class List extends PureComponent {
     renderItem = ({item}) => {
         return (
             <ChannelItem
+                ref={(ref) => {
+                    this.itemRefs[item] = ref;
+                }}
                 channelId={item}
                 onSelectChannel={this.onSelectChannel}
+                onLongPress={() => this.onPreview(item)}
```

Comment:
Could he have these take `onPreview` directly and then they know to call it with the channelId? That would save us from creating a new function here.

Reply:
To do this I’m going to have to try doing the preview from within the ChannelItem component instead and for that I’ll need to use Animated.View instead of a regular View cause the navigation API expect the ref to be a Component and a View is not a component